File: 백준/Bronze/14928. 큰 수 （BIG）/큰 수 （BIG）.java
import java.io.*;

class Main {
    static final int MOD = 2000_0303;
    
    public static void main(String[] args) throws IOException {
        int c, n = System.in.read() & 15;
        while ((c = System.in.read()) > 32) {
            n = ((n << 3) + (n << 1) + (c & 15)) % MOD;
        }
        System.out.print(n);
    }
}

File: 백준/Bronze/14928. 큰 수 （BIG）/큰 수 （BIG）.rs
use std::io::{self, Read};

const MOD: i32 = 2000_0303;

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).unwrap();
    let numbers = buffer.as_str().trim();
    
    let mut output: i32 = 0;
    for character in numbers.chars() {
        let c = character.to_digit(10).unwrap() as i32;
        output = ((output * 10) + c) % MOD;
    }
    println!("{}", output);
}